include/vctr/Expressions/BasicMath/CompareMask.h

```diff
@@ -0,0 +1,159 @@
+
+    VCTR is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License version 3 for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    version 3 along with VCTR.  If not, see <https://www.gnu.org/licenses/>.
+  ==============================================================================
+*/
+
+namespace vctr::expressions
+{
+
+// TODO These are only defined on Windows
+// Comparators
+enum class CompareOp : int
+{
+    Less = _CMP_LT_OQ,           // <
+    LessOrEqual = _CMP_LE_OQ,    // <=
+    Greater = _CMP_GT_OQ,        // >
+    GreaterOrEqual = _CMP_GE_OQ, // >=
+    Equal = _CMP_EQ_OQ,          // ==
+    NotEqual = _CMP_NEQ_OQ       // !=
+};
+
+template <CompareOp Op, class T>
+VCTR_FORCEDINLINE constexpr bool scalarCompare (const T& a, const T& b) noexcept
+{
+    if constexpr (Op == CompareOp::Less)
+        return a < b;
+    if constexpr (Op == CompareOp::LessOrEqual)
+        return a <= b;
+    if constexpr (Op == CompareOp::Greater)
+        return a > b;
+    if constexpr (Op == CompareOp::GreaterOrEqual)
+        return a >= b;
+    if constexpr (Op == CompareOp::Equal)
+        return a == b;
+    if constexpr (Op == CompareOp::NotEqual)
+        return a != b;
+    return false;
+}
+
+template <size_t extent, class SrcAType, class SrcBType, CompareOp Op>
+class CompareVecVecToMask : ExpressionTemplateBase
+{
+public:
+    using value_type = bool;
+    using AValue = ValueType<SrcAType>;
+    using BValue = ValueType<SrcBType>;
+
+    static_assert (std::is_floating_point_v<AValue> && std::is_floating_point_v<BValue>,
+                   "CompareVecVecToMask currently supports floating-point element types");
+
+    VCTR_COMMON_BINARY_VEC_VEC_EXPRESSION_MEMBERS (CompareVecVecToMask, a, b)
+
+    // per-element access
+    VCTR_FORCEDINLINE constexpr value_type operator[] (size_t i) const
+    {
+        return scalarCompare<Op> (static_cast<AValue> (a[i]), static_cast<BValue> (b[i]));
+    }
+
+    //==============================================================================
+    // AVX Implementation
+    VCTR_FORCEDINLINE VCTR_TARGET ("avx") void prepareAVXEvaluation() const
+    requires has::prepareAVXEvaluation<SrcAType> && has::prepareAVXEvaluation<SrcBType>
+    {
+        a.prepareAVXEvaluation();
+        b.prepareAVXEvaluation();
+    }
+
+    //VCTR_FORCEDINLINE VCTR_TARGET ("avx") AVXRegister<value_type> getAVX (size_t i) const
+    //{
+    //    const __m256 compareResult = _mm256_cmp_ps (a.getAVX (i).value, b.getAVX (i).value, int (Op));
+    //    const int mask = _mm256_movemask_ps (compareResult);
+    //
+    //    // write bool output vector
+    //    for (unsigned i = 0; i < lanes; ++i)
+    //        dst[registerIndex + i] = ((mask >> i) & 1) != 0;
+    //}
```

Comment:
Ok, here the pattern I used before with `getAVX()` didn't work out, because the return type is not the same as the inputs? I used the `evalNextVectorOpInExpressionChain()` below instead to run over the entire buffer